test/local/account_routes.js

```diff
@@ -810,21 +810,48 @@ test('/account/login', function (t) {
         mockMailer.sendNewDeviceLoginNotification.reset()
       })
     }, t)
+
+    test('unverified account does not get any confirmation emails', function (t) {
+      config.signinConfirmation.supportedClients = [ 'fx_desktop_v3' ]
+      mockRequest.payload.email = 'test@mozilla.com'
+      mockDB.emailRecord = function () {
+        return P.resolve({
+          authSalt: crypto.randomBytes(32),
+          data: crypto.randomBytes(32),
+          email: mockRequest.payload.email,
+          emailVerified: false,
+          kA: crypto.randomBytes(32),
+          lastAuthAt: function () {
+            return Date.now()
+          },
+          uid: uid,
+          wrapWrapKb: crypto.randomBytes(32)
+        })
+      }
+
+      return runTest(route, mockRequest, function (response) {
+        t.equal(mockMailer.sendNewDeviceLoginNotification.callCount, 0, 'mailer.sendNewDeviceLoginNotification was not called')
+        t.equal(mockMailer.sendVerifyLoginEmail.callCount, 0, 'mailer.sendVerifyLoginEmail was not called')
+        t.equal(response.verificationMethod, 'email', 'verificationMethod is email')
+        t.equal(response.verificationReason, 'signup', 'verificationReason is signup')
+      })
+    }, t)
```

Comment:
Should there also be a test for unverified emails still receiving the account verified email when the user verifies the signup email?